components/script/dom/htmlimageelement.rs

```diff
@@ -432,10 +433,14 @@ impl HTMLImageElement {
             _ => {},
         };
 
-        let map = self.upcast::<Node>()
-                      .following_siblings()
-                      .filter_map(Root::downcast::<HTMLMapElement>)
-                      .find(|n| n.upcast::<Element>().get_string_attribute(&LocalName::from("name")) == last);
+        let parent = match self.upcast::<Node>().GetParentNode() {
+            None => return None,
+            Some(p) => p,
+        };
+
+        let map = parent.children()
+                        .filter_map(Root::downcast::<HTMLMapElement>)
+                        .find(|n| n.upcast::<Element>().get_string_attribute(&LocalName::from("name")) == last);
```

Comment:
That doesn't seem right to me. The whole method can still panic if there was no `HTMLMapElement` in the image's parent's children.